DataFormats/Detectors/TRD/src/AngularResidHistos.cxx

```diff
@@ -31,14 +31,14 @@ bool AngularResidHistos::addEntry(float deltaAlpha, float impactAngle, int chamb
   // add entry for given angular residual
   // returns 0 in case of success (impact angle is in valid range)
   int chamberOffset = chamberId * NBINSANGLEDIFF;
-  if (std::fabs(impactAngle) >= MAXIMPACTANGLE) {
-    LOG(debug) << "Under-/overflow entry detected for impact angle " << impactAngle;
-    return 1;
-  } else {
+  if (std::fabs(impactAngle) < MAXIMPACTANGLE) {
     int iBin = (impactAngle + MAXIMPACTANGLE) * INVBINWIDTH;
     mHistogramEntries[chamberOffset + iBin] += deltaAlpha;
     ++mNEntriesPerBin[chamberOffset + iBin];
     ++mNEntriesTotal;
+  } else {
+    LOG(debug) << "Under-/overflow entry detected for impact angle " << impactAngle;
```

Comment:
Not sure how meaningful logging nan here really is, maybe also catching before hand getting the Snp which caused this?
Something like this:
```
      float trkAngle = o2::math_utils::asin(trkWork.getSnp()) * TMath::RadToDeg();
      if (std::isnan(trkAngle)) {
        LOG(debug) << "Caught NaN value form asin domain error: Snp=" << trkWork.getSnp();
      }
```

Reply:
The check for nan is in another place as I wrote above. Here I actually just check if the impact angle is in the valid range, which it can easily not be for more curved tracks with lower pT. The change I hope just gives a better protection from crashes, it's not meant for debugging the actual issue